com.microsoft.mrtk.spatialmanipulation/Utilities/SolverHandler.cs

```diff
@@ -400,7 +391,8 @@ public void RegisterSolver(Solver solver)
             if (!solvers.Contains(solver))
             {
                 solvers.Add(solver);
-                updateSolversList = true;
```

Comment:
I think this was originally patterned the other way so that multiple solvers being registered in a single frame wouldn't cause multiple queries and (I think?) allocations. Instead, the handler would be marked to update its list once before updating them in `LateUpdate`.

All that said, I don't think I'm against this change if you were seeing issues with the other pattern!

Reply:
Did an additional minor change to just get rid of the Union and Intersect calls all together. The way that the solver class is set up, they can only exist on the same object as the SolverHandler, so just using `GetComponents(solvers)` should be sufficient. I think we might not have wanted the solver handler and the solvers themselves to both be monobehaviors, but I think a rewrite would be more cumbersome than improved code quality it would provide.

Reply:
Made some changes to better capture this behavior. This PR does introduce the change where the solverhandler will no longer manage solvers which are not attached to the same gameobject, but because solvers have a `RequiresComponent(SolverHandler)`, I think this change ultimately helps protect the intended pattern.

Making a work item to revisit and better streamline the solver system to be more user friendly and easier to build ontop of.